src/main/java/org/spongepowered/api/locale/RemoteDictionary.java

```diff
@@ -0,0 +1,133 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.locale;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+/**
+ * Represents a {@link Dictionary} with a remote source.
+ */
+public interface RemoteDictionary extends Dictionary {
+
+    /**
+     * Returns an {@link InputStream} for the remote source of this Dictionary.
+     *
+     * @return Source of dictionary
+     * @throws IOException
+     */
+    InputStream getSource(Locale locale) throws Exception;
+
+    /**
+     * Returns an {@link InputStream} for the remote source of this Dictionary.
+     *
+     * @return Source of dictionary
+     * @throws IOException
+     */
+    default InputStream getSource() throws Exception {
+        return getSource(getDefaultLocale());
+    }
+
+    /**
+     * Class that resolves an {@link InputStream} source for a given {@link Locale}.
+     */
+    class SourceResolver {
+
+        Callable<InputStream> primary;
+        final Map<Locale, List<Callable<InputStream>>> sources = new HashMap<>();
+
+        SourceResolver() {
+        }
+
+        /**
+         * Sets the primary or "default" resolver.
+         *
+         * @param primary Resolver
+         * @return This resolver
+         */
+        public SourceResolver primary(Callable<InputStream> primary) {
+            this.primary = checkNotNull(primary, "primary");
+            return this;
+        }
+
+        /**
+         * Adds a {@link Callable} to supply an {@link InputStream} for the
+         * given {@link Locale}.
+         *
+         * @param locale Locale to supply for
+         * @param callable Callable to supply InputStream
+         * @return This resolver
+         */
+        public SourceResolver add(Locale locale, Callable<InputStream> callable) {
+            checkNotNull(locale, "locale");
+            checkNotNull(callable, "callable");
+            List<Callable<InputStream>> list = this.sources.get(locale);
+            if (list == null) {
+                list = new ArrayList<>();
+            }
+            list.add(callable);
+            this.sources.put(locale, list);
+            return this;
+        }
+
+        /**
+         * Resolves the {@link InputStream} source for the given
+         * {@link Locale}.
+         *
+         * @param locale Locale to resolve source for
+         * @return Optional InputStream. Empty if unresolved.
+         * @throws Exception
+         */
+        public Optional<InputStream> resolve(Locale locale) throws Exception {
+            checkNotNull(locale, "locale");
```

Comment:
Again, magic this magic string is kind of scattered all over the place. If you can find somewhere convenient to put it, that'd be cool.